### ISSUE TITLE
Allowing incorrect return type when doing arithmetic with Literal-bounded type

### ISSUE THREAD

**Issue by lavirlifiliol** [microsoft/pyright#9090]
**Describe the bug**
A generic `T` bounded by a typing.Literal with multiple possible values can be instantiated into a typing.Literal with any subset of the possible values. However, Pyright occasionally treats it as if it were a fixed type that only ever ends up instatiated as the full type in the bound.

**Code or Screenshots**
```py
from typing import Literal
type Foo = Literal[1, 0, -1]

# does type check, incorrectly
def foo[T: Foo](bar: T) -> T:
    return -bar
m: Literal[-1] = -1
m = foo(m) # type of m is Literal[-1], but m=1 here

# does not type check, correctly
def baz[T: tuple[Foo]](bar: T) -> T:
    return -bar[0],
```
**VS Code extension or command-line**
tested on pyright-play.net, version 1.1.382

**Comment by erictraut** [comment]
Thanks for the bug report and the clear and concise code sample. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.383.
